### DISCUSSION TITLE
Compiling 5.1 without Pulseaudio

### DISCUSSION THREAD

**Discussion by farsil** [stepmania/stepmania#2218] [Q&A] [answered]
Hello,

I want to compile Stepmania 5.1 on Tiny Core Linux, an OS suitable for embedded systems. I plan to build a custom linux image for my cab.

I have noticed that 5.1 doesn't compile anymore without Pulseaudio. I don't like Pulseaudio for this use case, because it brings in a ridiculous amount of dependencies. Is it a core component of 5.1 or there's a way around it?

Thank you!

**Comment by geefr** [comment]
Builds fine without pulse for me, you should provide your cmake parameters, build logs, and any error messages.
(And assuming you mean the 5_1-new branch by '5.1')

You will need to have one audio system enabled - If pulse is turned off then either alsa or jack should be on.
(Though I think if you really want, you can build with no audio support..)

**Comment by farsil** [comment, marked as answer]
Works if I set `WITH_PULSEAUDIO` to `off` in the cmake parameters.

```sh
$ cd Build
$ cmake -DCMAKE_BUILD_TYPE=Debug -DWITH_PULSEAUDIO=off ..
```
This is not necessary in the 5.1 beta 2 source, but it is if you compile from the current `5_1-new` branch.
